src/test/java/org/cactoos/http/HtUpgradeWireTest.java

```diff
@@ -0,0 +1,167 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.http;
+
+import java.io.IOException;
+import java.io.InputStream;
+import org.cactoos.Input;
+import org.cactoos.text.TextOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.hamcrest.core.IsInstanceOf;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.takes.Request;
+import org.takes.Response;
+import org.takes.Take;
+import org.takes.http.FtRemote;
+import org.takes.rs.RsWithStatus;
+import org.takes.tk.TkText;
+
+/**
+ * Test for {@link HtUpgradeWire}. Must test if the {@link Wire} returned
+ * after an 101 status code is an {@link HtSecureWire}.
+ * @author Paulo Lobo (pauloeduardolobo@gmail.com)
+ * @version $Id$
+ * @since 0.1
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class HtUpgradeWireTest {
+
+    /**
+     * Tests if the wire of the response of an 101 code is an
+     * {@link HtSecureWire}.
+     * @throws Exception If Something goes wrong.
+     */
+    @Test
+    @Ignore("HtUpgradeWire implementation is not ready yet.")
+    public void testUpgrade() throws Exception {
+        new FtRemote(new TkAlways101Mock(new TkText("Hello, world!"))).exec(
+            home -> MatcherAssert.assertThat(
+                "Could not upgrade wire",
+                new ResponseWrap(
+                    new HtUpgradeWire(
+                        new HtWire(
+                            home.getHost(), home.getPort()
+                        )
+                    ),
+                    home.getHost()
+                ).wire(),
+                new IsInstanceOf(HtSecureWire.class)
+            )
+        );
+    }
+
+    /**
+     * Test of {@link HtUpgradeWire} just to suit coverage standards.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public void testHtUpgrade() throws Exception {
+        new FtRemote(new TkText("Upgraded wire")).exec(
+            home -> MatcherAssert.assertThat(
+                "Upgrade wire not found",
+                new TextOf(
+                    new HtResponse(
+                        new HtUpgradeWire(
+                            new HtWire(
+                                home.getHost(),
+                                home.getPort()
+                            )
+                        ),
+                        home.getHost()
+                    )
+                ).asString(),
+                Matchers.containsString("HTTP/1.1 200")
+            )
+        );
+    }
+
+    /**
+     * Wrap for response which allows access to its wire.
+     */
+    private class ResponseWrap implements Input {
+
+        /**
+         * Origin response.
+         */
+        private final HtResponse origin;
```

Comment:
Why is this called `origin`, and not a `response`? `origin` is fimilar to decorators, but `HtResponse` is not `Input`.

Reply:
@VsSekorin Didn't unsderstood your remark, `HtResponse` is an `Input` indeed.

Reply:
My bad. Thanks